### ISSUE TITLE
Wear OS app crashes when bluetooth is abruptly disconnected 

### ISSUE THREAD

**Issue by szescxz** [jamie-mh/AuthenticatorPro#53]
```
09-27 09:52:40.688  7114  7114 I MonoDroid: UNHANDLED EXCEPTION:
09-27 09:52:41.205  7114  7114 I MonoDroid: Android.Gms.Common.Apis.ApiException: 4000: TARGET_NODE_NOT_CONNECTED
09-27 09:52:41.205  7114  7114 I MonoDroid:   at AuthenticatorPro.WearOS.Activity.MainActivity.Refresh () [0x000b6] in <a6a47d639c2c4c28a29bfc4a049dbc50>:0
09-27 09:52:41.205  7114  7114 I MonoDroid:   at AuthenticatorPro.WearOS.Activity.MainActivity.OnCreate (Android.OS.Bundle bundle) [0x0039e] in <a6a47d639c2c4c28a29bfc4a049dbc50>:0
09-27 09:52:41.205  7114  7114 I MonoDroid:   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in <5e5e1d94f16a41abbcb4038adc765ccd>:0
09-27 09:52:41.205  7114  7114 I MonoDroid:   at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <b0945a35bf934a47bf1779cc47fc7a0b>:0
09-27 09:52:41.205  7114  7114 I MonoDroid:   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <b0945a35bf934a47bf1779cc47fc7a0b>:0
09-27 09:52:41.205  7114  7114 I MonoDroid:   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in <b0945a35bf934a47bf1779cc47fc7a0b>:0
09-27 09:52:41.206  7114  7114 I MonoDroid:   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.8(intptr,intptr)
09-27 09:52:41.206  7114  7114 I MonoDroid:   --- End of managed Android.Gms.Common.Apis.ApiException stack trace ---
09-27 09:52:41.206  7114  7114 I MonoDroid: com.google.android.gms.common.api.ApiException: 4000: TARGET_NODE_NOT_CONNECTED
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.common.internal.ApiExceptionUtil.fromStatus(Unknown Source:14)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.common.internal.zai.zaf(Unknown Source:0)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.common.internal.zaj.onComplete(Unknown Source:32)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.common.api.internal.BasePendingResult.zaa(Unknown Source:79)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.common.api.internal.BasePendingResult.setResult(Unknown Source:41)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.common.api.internal.BaseImplementation$ApiMethodImpl.setResult(Unknown Source:2)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.wearable.internal.zzgm.zza(Unknown Source:4)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.wearable.internal.zzhe.zza(Unknown Source:13)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.wearable.internal.zzel.dispatchTransaction(Unknown Source:265)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at com.google.android.gms.internal.wearable.zzb.onTransact(Unknown Source:22)
09-27 09:52:41.206  7114  7114 I MonoDroid:     at android.os.Binder.execTransact(Binder.java:731)
09-27 09:52:41.206  7114  7114 I MonoDroid:
09-27 09:52:41.258  7114  7114 D AndroidRuntime: Shutting down VM
09-27 09:52:41.284  7114  7114 E AndroidRuntime: FATAL EXCEPTION: main
09-27 09:52:41.284  7114  7114 E AndroidRuntime: Process: me.jmh.authenticatorpro, PID: 7114
09-27 09:52:41.284  7114  7114 E AndroidRuntime: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:503)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
09-27 09:52:41.284  7114  7114 E AndroidRuntime: Caused by: java.lang.reflect.InvocationTargetException
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        ... 1 more
09-27 09:52:41.284  7114  7114 E AndroidRuntime: Caused by: com.google.android.gms.common.api.ApiException: 4000: TARGET_NODE_NOT_CONNECTED
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.common.internal.ApiExceptionUtil.fromStatus(Unknown Source:14)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.common.internal.zai.zaf(Unknown Source:0)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.common.internal.zaj.onComplete(Unknown Source:32)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.common.api.internal.BasePendingResult.zaa(Unknown Source:79)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.common.api.internal.BasePendingResult.setResult(Unknown Source:41)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.common.api.internal.BaseImplementation$ApiMethodImpl.setResult(Unknown Source:2)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.wearable.internal.zzgm.zza(Unknown Source:4)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.wearable.internal.zzhe.zza(Unknown Source:13)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.wearable.internal.zzel.dispatchTransaction(Unknown Source:265)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at com.google.android.gms.internal.wearable.zzb.onTransact(Unknown Source:22)
09-27 09:52:41.284  7114  7114 E AndroidRuntime:        at android.os.Binder.execTransact(Binder.java:731)
```
My steps to reproduce:
1. Ensure the watch app is connected to the phone app (no blue "crossed-out cloud" icon)
2. Turn off watch's bluetooth
3. Launch watch app immediately

I'd also like to report that the Wear OS app doesn't seem to work well on Wear OS China, as I have to force stop "Wear OS by Google" phone app to make Authenticator Pro's watch app recognize and sync with the phone app during initial setup. Tested on Mi Watch.

**Comment by jamie-mh** [comment]
Hi @szescxz ,

Indeed, I can reproduce this crash. I will look into it.

As for the app not working well on Wear OS China, I don't think there is much I can do about that. There is no code in the app that is specific to the standard version of Wear OS, this might be just a bug with the firmware on the watch.

**Comment by jamie-mh** [comment]
Fixed in eff8b4e.
It seems there is a bug in Wear OS, in which for a while after disconnecting from the phone, the watch isn't notified.
